controllers/cinder_controller.go

```diff
@@ -341,16 +344,41 @@ func (r *CinderReconciler) reconcileInit(
 func (r *CinderReconciler) reconcileNormal(ctx context.Context, instance *cinderv1beta1.Cinder, helper *helper.Helper) (ctrl.Result, error) {
 	r.Log.Info(fmt.Sprintf("Reconciling Service '%s'", instance.Name))
 
-	// If the service object doesn't have our finalizer, add it.
-	controllerutil.AddFinalizer(instance, helper.GetFinalizer())
-	// Register the finalizer immediately to avoid orphaning resources on delete
-	if err := r.Update(ctx, instance); err != nil {
+	if !controllerutil.ContainsFinalizer(instance, helper.GetFinalizer()) {
```

Comment:
Yes, these are riders along with this PR to fix a serious problem.  Please see [1].  I suppose this could be done in a separate PR, but I wanted to do it while it was fresh on my mind.

[1] https://github.com/openstack-k8s-operators/ovn-operator/pull/1#discussion_r997014701

Reply:
I'm ok doing it in the same PR, but it should be a different commit within the PR so when we look at the git history we can see in its commit message why this change was made.

Reply:
I will change those modifications into a separate commit before we merge.  I think we had been using "squash and merge" in OSPdO and that has somehow hardwired my brain into believing we were doing the same here.

Reply:
Thank you. :-)
I'd rather we move away from squashing in the cinder-operator and from oneliner commit messages like the k8s community does.
The source of truth should be the git history, not the PR message or discussion, but this is a bigger discussion.